src/psyclone/psyGen.py

```diff
@@ -5364,13 +5556,110 @@ def _if_stmt_handler(self, node, parent):
         :returns: PSyIR representation of node
         :rtype: :py:class:`psyclone.psyGen.IfBlock`
         '''
-        ifblock = IfBlock(parent=parent)
+        ifblock = IfBlock(parent=parent, annotation='was_single_stmt')
         self.process_nodes(parent=ifblock, nodes=[node.items[0]],
                            nodes_parent=node)
-        self.process_nodes(parent=ifblock, nodes=[node.items[1]],
+        ifbody = Schedule(parent=ifblock)
+        ifblock.addchild(ifbody)
+        self.process_nodes(parent=ifbody, nodes=[node.items[1]],
                            nodes_parent=node)
         return ifblock
 
+    def _case_construct_handler(self, node, parent):
+        '''
+        Transforms an fparser2 Case_Construct to the PSyIR representation.
+
+        :param node: node in fparser2 tree.
+        :type node: :py:class:`fparser.two.Fortran2003.Case_Construct`
+        :param parent: Parent node of the PSyIR node we are constructing.
+        :type parent: :py:class:`psyclone.psyGen.Node`
+        :returns: PSyIR representation of node
+        :rtype: :py:class:`psyclone.psyGen.IfBlock`
+        :raises InternalError: If the fparser2 tree has an unexpected \
+            structure.
+        :raises NotImplementedError: If the fparser2 tree contains an \
+            unsupported structure and should be placed in a CodeBlock.
+        '''
+        from fparser.two import Fortran2003
+        # Check that the fparser2 parsetree has the expected structure
+        if not isinstance(node.content[0], Fortran2003.Select_Case_Stmt):
+            raise InternalError(
+                "Failed to find opening case statement in: "
+                "{0}".format(str(node)))
+        if not isinstance(node.content[-1], Fortran2003.End_Select_Stmt):
+            raise InternalError(
+                "Failed to find closing case statement in: "
+                "{0}".format(str(node)))
+
+        # Search for all the CASE clauses in the Case_Construct
+        clause_indices = []
+        selector = None
+        for idx, child in enumerate(node.content):
+            child._parent = node  # Retrofit parent info
+            if isinstance(child, Fortran2003.Select_Case_Stmt):
+                selector = child.items[0]
+            if isinstance(child, (Fortran2003.Case_Stmt,
+                                  Fortran2003.End_Select_Stmt)):
+                # Case Default and value Ranges not supported yet
+                if 'DEFAULT' in str(child):
+                    raise NotImplementedError("Case Default Statement")
+                elif ':' in str(child):
+                    raise NotImplementedError("Case Value Range Statement")
+                else:
```

Comment:
pylint says this `else:` is unnecessary (because it follows a `raise`)

Reply:
deleted